src/DiscoveryClient.coffee

```diff
@@ -6,40 +6,63 @@ DiscoveryLongPoller = require "./DiscoveryLongPoller"
 ServerList          = require "./ServerList"
 Utils               = require "./Utils"
 Promise             = require "bluebird"
+async               = require "async"
+
 
 class DiscoveryClient
 
-  constructor:(@host, @options)->
+  constructor:(@host, @serviceName, @regionsConfig, @options) ->
+    if not @regionsConfig || not @regionsConfig.discoverRegion || not @regionsConfig.announceRegions
+      throw new Error "Must pass in regionsConfig param with at least one region for discovery apiv2"
+
+
+    if @host.indexOf('*') < 0
+      throw new Error 'Must have a wildcard to replace with region name in host!'
```

Comment:
I personally prefer being more explicit. Is it ok to specify the host in each of the regionsConfig instead?